### ISSUE TITLE
View visibility changes

### ISSUE THREAD

**Issue by TheBizzle** [NetLogo/NetLogo#681]
Alan Isaac writes:

> One can hide the Graphics Window by setting size width to a negative number, but this is undocumented and clunky.  Please consider the following.
>
> 1. A `View` menu with a hide/show switch for the Graphics Window. (Some `Tools`, such as Hiding/showing the Command Center, could be moved there as well.)
>
> 2. A Boolean primitive to accomplish this (e.g., `display-graphics-window`). This should be usable in the `startup` procedure.
>
> More radically, since some models do not need a graphics window, one might introduce a Boolean primitive to allow disabling its creation (in the startup procedure).

**Comment by TheBizzle** [comment]
I agree with the general sentiment of this idea.  I think it's odd that NetLogo forces models to have a View, because it's not true that all models _use_ the View—the vast majority, yes, but not all.  Some examples of models that don't need a View are those concerned with just plots, just textual output (be it into the Command Center or some other widget), or simply monitoring readings from hardware devices.

Regarding adding primitives for controlling the visibility of the View, I have some opposition to that—albeit not strong opposition.  My opinion is that we should try to keep the language primitives focused on the objects that live _within_ the model (e.g. agentsets and variables—stuff used for computer new model states), rather than having primitives for introspecting the model's settings.

Instead, I think that the View should be there by default (like it currently is), but that users should just be able to delete the View widget like any other widget, and be able to have at most one View in their model (like how we treat Output widgets).  If a View is present, we draw to it on tick (or when it is created), otherwise not.

It's not clear to me what use case(s) exist for hiding/showing the View dynamically throughout the run of the same model.

**Comment by nicolaspayette** [comment]
Related: #326, #409.

**Comment by SethTisue** [comment]
if this is added, I'd suggest making it a checkbox in the Settings dialog.

**Comment by TheBizzle** [comment]
Bumping and assigning to the Hexy milestone.

Rather than relying on kludges with negative patch sizes or hiding the View behind other widgets, I continue to believe that we should embrace the fact that the View is just another widget, and it should be addable/removable like other widgets.

It can—and probably should—still be there by default when a "blank" model is started, but I think it would make sense to be able to remove it like other widgets, and to add it back later using the standard widget-adding controls.  A model should only be permitted to have one view widget at most (like how the output widget is handled).

**Comment by mrerrormessage** [comment]
Also related #542

**Comment by SethTisue** [comment]
I like Jason's suggestion better than mine.

**Comment by mrerrormessage** [comment]
This will have to be deferred until after the new model format is ready. This cannot reasonably be done within the existing ".nlogo" file format. We could use some hacky method, but then we'd produce ".nlogo" files which were very likely to be backwards-incompatible.

**Comment by nicolaspayette** [comment]
A few models that don't need a view resize it to what used to be the height of the control strip. Since #930, however, there is no more control strip, and models that did that now look weird. For example, Simple Genetic Algorithm:

![untitled](https://cloud.githubusercontent.com/assets/908754/17940664/0d302e24-6a28-11e6-9076-3e4f1d5f19d0.png)

No big deal, but maybe another point in favor of making the view truly hideable.
